Breaking change: close() as instance method

gverni opened the pull request:
I noticed the TODO comment on: https://github.com/sweetalert2/sweetalert2/blob/f76808ef4876a2f0602a4da04a07223e0e8c5821/src/instanceMethods/_main.js#L46

and I took the opportunity of the fix for issue #919 to refactor `close()` as instance method. 

I also removed the aliases `closeModal()` and `closeToast()` that were not used / documented anywhere. 

@limonte @zenflow let me know if I _went too far_ and i'm happy to implement to implement the fix for `close()` as static method.

zenflow commented:
@gverni Yup, I think we want this change for the upcoming major release. Note that it is a breaking change to the API and thus not technically just a refactor. 

We need to do this for all the static methods that make more sense to be instance methods.. see https://github.com/sweetalert2/sweetalert2/issues/1373

limonte commented:
![image](https://user-images.githubusercontent.com/6059356/51077696-afbe8800-16b2-11e9-97ea-6b798e45eff1.png)

Congrats, everyone!

gverni commented:
:-) Congratulations!!!

On Sat, 12 Jan 2019, 19:40 Limon Monte, <notifications@github.com> wrote:

> [image: image]
> <https://user-images.githubusercontent.com/6059356/51077696-afbe8800-16b2-11e9-97ea-6b798e45eff1.png>
>
> Congrats, everyone!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sweetalert2/sweetalert2/pull/1379#issuecomment-453775497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQTDdLHkI0tKEi1DUwHlPGvvz7qBWpCFks5vCjpKgaJpZM4Z7_nT>
> .
>

gverni commented:
Hi @limonte sorry it took me so long, but I was trying to find the most correct way to achieve: 

> resolve the promise with {} (no value or dismiss)

What I did, is to add `resolve` and `reject` into a new `innerProps.innerFunctions` object. To be honest, I'd rather create a new `innerMethods` object and add those to that, if you like this approach. 

Alternatively, I tried also to save reference to `reject` / `resolve` in one of the properties of `Swal`, but this will expose these methods, and I personally did not like it (though it obviously works). 

To be honest, I'm not really happy with that, but I couldn't think of any better way of achieving that goal. Feel free to let me know if there is one.

gverni commented:
The failure on CI is obviously due to the fact that i'm always resolving the promise at the end of `close()`. Before fixing that, I just want to be sure i'm on the right path with this approach.

gverni commented:
I really didn't like to use `privateProps` to hold the promise's `resolve` / `reject`. So I added a `privateMethods` object to hold these (and future) private methods. 

I also moved the logic for handling `resolve` / `reject` directly within `close()` method

gverni commented:
Given #1383 I can simplify even more the prototype of the `close()` and keep only the `resolveValue`

limonte commented:
> Given #1383 I can simplify even more the prototype of the close() and keep only the resolveValue

That's a possible change, yeah. But let's agree on it before doing any changes on this PR.

gverni commented:
@limonte @zenflow I think i'm done with this PR. Feel free to review it.

limonte commented on test/qunit/methods/close.js:
```diff
@@ -0,0 +1,71 @@
+QUnit.test('close() method', (assert) => {
+  const done = assert.async()
+
+  // create a modal with an onAfterClose callback
```
all "create a modal with an onAfterClose callback" aren't needed, please remove here and below

gverni replied:
Resolved by https://github.com/sweetalert2/sweetalert2/pull/1379/commits/cea3b81f0a1470bfdeb8ff18ffce92ef080b9c46

gverni commented:
thanks @limonte!

| Do I understand correctly that this PR doesn't contain any breaking changes?

The _breaking change_ is in the fact that now if you call `Swal.close()` it does resolve the promise (with `{}`), whereas before it didn't. Do you agree?

limonte commented:
:tada: This has been shipped in version 8.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/sweetalert2)
- [GitHub release](https://github.com/sweetalert2/sweetalert2/releases/tag/v8.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
